spec/loader_spec.js

```diff
@@ -8,38 +8,33 @@ describe('loader', function() {
   describe('#load', function() {
     describe('When the path ends in .mjs', function () {
       it('loads the file as an es module', async function () {
-        const requireShim = jasmine.createSpy('requireShim');
-        let resolve;
-        const importPromise = new Promise(function (res) {
-          resolve = res;
-        });
+        const requireShim = jasmine.createSpy('requireShim')
+          .and.callFake(requireESM);
         const importShim = jasmine.createSpy('importShim')
-          .and.returnValue(importPromise);
+          .and.returnValue(Promise.resolve());
         const loader = new Loader({requireShim, importShim});
 
-        const loaderPromise = loader.load('./foo/bar/baz.mjs');
+        await expectAsync(loader.load('./foo/bar/baz.mjs')).toBeResolved();
```

Comment:
Had to wait for completion here, because `importShim` is not called immediately anymore. I have to wait for the `requireShim` to error, before `importShim` is even called.